google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/changestream/ChangeStreamStateMachine.java

```diff
@@ -356,161 +351,138 @@ State handleMod(ReadChangeStreamResponse.DataChange dataChange, int index) {
-   *       mods to expect.
-   *   <dd>{@link ChangeStreamStateMachine#AWAITING_CELL_VALUE}. Current mod is the first chunk of a
-   *       chunked SetCell.
-   *   <dd>{@link ChangeStreamStateMachine#AWAITING_STREAM_RECORD_CONSUME}. Current mod is the last
-   *       mod of the current logical mutation.
+   *   <dd>{@link ChangeStreamStateMachine#AWAITING_NEW_DATA_CHANGE}. All mods from the current
+   *       DataChange are added, and we have more mods to expect.
+   *   <dd>{@link ChangeStreamStateMachine#AWAITING_STREAM_RECORD_CONSUME}. Current DataChange is
+   *       the last DataChange of the current logical mutation.
    * </dl>
    */
-  private final State AWAITING_NEW_MOD =
+  private final State AWAITING_NEW_DATA_CHANGE =
       new State() {
         @Override
         State handleHeartbeat(ReadChangeStreamResponse.Heartbeat heartbeat) {
           throw new IllegalStateException(
-              "AWAITING_NEW_MOD: Can't handle a Heartbeat in the middle of building a ChangeStreamMutation.");
+              "AWAITING_NEW_DATA_CHANGE: Can't handle a Heartbeat in the middle of building a ChangeStreamMutation.");
         }
 
         @Override
         State handleCloseStream(ReadChangeStreamResponse.CloseStream closeStream) {
           throw new IllegalStateException(
-              "AWAITING_NEW_MOD: Can't handle a CloseStream in the middle of building a ChangeStreamMutation.");
+              "AWAITING_NEW_DATA_CHANGE: Can't handle a CloseStream in the middle of building a ChangeStreamMutation.");
         }
 
         @Override
-        State handleMod(ReadChangeStreamResponse.DataChange dataChange, int index) {
-          validate(
-              0 <= index && index <= dataChange.getChunksCount() - 1,
-              "AWAITING_NEW_MOD: Index out of bound.");
-          ReadChangeStreamResponse.MutationChunk chunk = dataChange.getChunks(index);
-          Mutation mod = chunk.getMutation();
-          // Case 1: SetCell
-          if (mod.hasSetCell()) {
-            // Start the Cell and delegate to AWAITING_CELL_VALUE to add the cell value.
-            Mutation.SetCell setCell = chunk.getMutation().getSetCell();
-            if (chunk.hasChunkInfo()) {
-              // If it has chunk info, it must be the first chunk of a chunked SetCell.
-              validate(
-                  chunk.getChunkInfo().getChunkedValueOffset() == 0,
-                  "AWAITING_NEW_MOD: First chunk of a chunked cell must start with offset==0.");
-              validate(
-                  chunk.getChunkInfo().getChunkedValueSize() > 0,
-                  "AWAITING_NEW_MOD: First chunk of a chunked cell must have a positive chunked value size.");
-              expectedTotalSizeOfChunkedSetCell = chunk.getChunkInfo().getChunkedValueSize();
-              actualTotalSizeOfChunkedSetCell = 0;
+        State handleDataChange(ReadChangeStreamResponse.DataChange dataChange) {
+          // Iterate over all mods.
```

Comment:
Did you explain what a mod is anywhere? Maybe explain a little bit here:

there are 3 types of mod:
setCell,
deleteFamily,
deleteFromColumn.

Reply:
Yeah I've updated the comments of `ChangeStreamStateMachine.State handleDataChange` to say that `A DataChange can have multiple mods, where each mod could be a DeleteFamily, a DeleteColumn, or a SetCell`